tensorflow/contrib/tensorrt/segment/segment.cc

```diff
@@ -25,18 +25,58 @@ limitations under the License.
 #include "tensorflow/core/graph/graph_constructor.h"
 #include "tensorflow/core/lib/core/errors.h"
 #include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/lib/strings/strcat.h"
 #include "tensorflow/core/platform/types.h"
 
 namespace tensorflow {
 namespace tensorrt {
 namespace segment {
-
+using ::tensorflow::strings::StrAppend;
 namespace {
 
-bool CanContractEdge(const tensorflow::Edge* edge,
-                     const tensorflow::Graph& graph) {
-  const tensorflow::Node* src = edge->src();
-  const tensorflow::Node* dst = edge->dst();
+bool check_cycles(const Graph* g, const Node* src,
```

Comment:
No. It takes TF graph not SimpleGraph. We were using that method before but we are not making copies of the graph for segmentation now.

Reply:
We were already using that when we were making copies of the graphs. But since this is prohibitively expensive,  we are using SimpleGraph now and the code you mentioned works only with TF Graph. check_cycles is pretty much a copy of the above code.